nemo_aligner/models/nlp/gpt/megatron_gpt_dpo_model.py

```diff
@@ -460,3 +463,7 @@ def get_ref_policy_logprobs(self, batch):
 
         # return in GPU, trainer needs to move to cpu
         return ref_log_probs
+
+class MegatronMambaDPOModel(MegatronMambaModel, MegatronGPTDPOModel): # @adithyare inherence order matters
```

Comment:
Why have you created new model classes? The logic to pad can just be switched on/off with a flag of some sort? It also seems like these new model classes don't have any new code?

Reply:
this is not for padding alone its also to support a new class of arch so we need this class.